tvaultuiv2/src/services/helper-function.js

```diff
@@ -40,19 +40,29 @@ export function findElementById(arr, id, nestingKey) {
     'Not found'
   );
 }
+export const findItemAndRemove = (arr, key, id) => {
+  if (arr?.length === 0) return;
+  const tempArr = [...arr];
+  const indexofItem = tempArr[0][key].findIndex((item) => item.id === id);
+  tempArr[0][key].splice(indexofItem, 1);
+  return tempArr;
+};
 
 export const findElementAndUpdate = (arr, parentId, item) => {
-  if (arr.length === 0) return;
+  if (arr?.length === 0) return;
   const tempArr = [...arr];
   const itemToUpdate = findElementById(tempArr, parentId, 'children');
   if (Array.isArray(item)) {
-    if (!itemToUpdate.children.length) {
+    if (!itemToUpdate?.children?.length) {
```

Comment:
For both the cases - If {Item is an array }or  else.... below statement is common.

**itemToUpdate.children = itemToUpdate?.children && [
      ...itemToUpdate.children,
      item,
    ];**

Reply:
if item is not an array then we are just adding item attribute. we need to check that condition